app/scripts/auditLog/models/auditLog.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var AuditLog = (function () {
+
+  var AuditLog = function AuditLog(data) {
+    this.loggerVersion = 1;
+    this.userType = data.hasOwnProperty('userType') ? data.userType : null;
+    this.userRef = data.hasOwnProperty('userRef') ? data.userRef : null;
+    this.operationType = data.hasOwnProperty('operationType') ? data.operationType : null;
+    this.dateTime = new Date();
```

Comment:
This is the date/time of an individual log entry, so it's the time of the change (creation, modification, deletion) of the record (brand, category, product).
Picking a log entry at random, it's the "dateTimeUtc" [here](https://console.firebase.google.com/project/end-slavery-now-test/database/end-slavery-now-test/data/auditLog/-Lnme0x8B1W_zeEATAUN).
I don't think "instantiatedAt" is an appropriate name, but we could change it to "eventDateTime" or maybe "createdAt"?
Do you think the name in the database ("dateTimeUtc") needs to change, too?

Reply:
Ahhh my bad! I misunderstood where it was used :) . Makes sense now! I like `eventDateTime` or `createdAt` - those both seem great to me. When I see the field on a record in firebase `dateTimeUtc` makes perfect sense, so I don't think that needs to change (also I have never seen the timezone referenced in the field name, but I love it! the number times I've had to dig to find out "what time zone is this stored in?" .... 🙃)

Reply:
That's not a problem :-)
I'm glad you like what I did with the timezone. So often I see someone create a database column or a class property and call it something like "height" or "waitTime", and I'm thinking, yes, I can see it's an int/float/whatever but what are the units?? Is it metres, inches, light years, milliseconds, days, what?? I have no idea without trawling through all the code, which is an absolute pain, like you say. Grrr!!! So I always make a point of putting the units, or as in this case, the timezone, in the name of the column/property. So I'd call them "HeightMetres" and "waitTimeMs".
We actually have createdAt and updatedAt on all our tables, and they're set automatically by the database.
I would have been happy with either `eventDateTime` or `createdAt`, but if you're happy with `dateTimeUtc`, I'd prefer to leave it as that as there are now records in firebase with that property name.

Reply:
Incidentally, the `loggerVersion` in the [audit log](https://github.com/elixirlabsinc/endslaverynow/blob/audit-log-phase-1/app/scripts/auditLog/models/auditLog.js#L6) model is there so that if anything changes, eg properties are added, old ones are removed or renamed, or datatypes change, we increment the version number with the release that those changes in, and the code that loads the records can process them properly according to the version number.
I've done it in one of my personal app, where it [loads a saved game](https://gitlab.com/pedanticantic/codebreaker/blob/master/src/assets/models/game.ts#L34) (you should have access). It basically takes what it's loaded, and systematically upgrades it to the current version. You'll notice that in the "upgrade to version 1", it introduces the version number (line 85)!